### PR TITLE
charm: add peer units to no-proxy config

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/86/

File: charm/hooks/config-changed [hunk 1]
```diff
@@ -42,12 +42,19 @@ def config_changed():
     else:
         app_config['location'] = config['location']
 
+    if config.get('http-proxy', None):
+        no_proxy = config.get('no-proxy', '')
+        for rel in hookenv.relations_of_type('candid'):
+            if len(no_proxy) > 0:
+                no_proxy += ','
+            no_proxy += rel.get('private-address')
```
Comment (rogpeppe) [inline charm/hooks/config-changed]: BTW, are you sure that a relation always has a private address set? Maybe worth being defensive and only adding to no_proxy if the address is non-empty?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/87/

**Comment by mhilton** [comment]
$$merge$$ :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**